contracts/LiquidDemocracy.sol

```diff
@@ -114,22 +119,38 @@ contract LiquidDemocracy {
         delegations[representee] = representative; // Delegate my own vote
 
         address nextRepresentative = representative;
-        address nextRepresentee = representee;
 
         // While there is voter that hasn't performed a delegation to another voter
-        // transitively add votes
+        // transitively add weight from their voting power
         while (delegations[nextRepresentative] != nextRepresentative) {
+            votersData[nextRepresentative].weight += votersData[representee].weight;
+            nextRepresentative = delegations[nextRepresentative];
+        }
 
-            votersData[nextRepresentative].weight += votersData[nextRepresentee].weight;
-            votersData[nextRepresentee].weight -= votersData[nextRepresentee].weight;
+        votersData[nextRepresentative].weight += votersData[representee].weight;
 
-            nextRepresentee = nextRepresentative;
-            nextRepresentative = delegations[nextRepresentative];
+    }
 
-        }
+    /**
+    * @notice Revoke the delegation and voting power already delegated
+    * @param from revoke delegations on this voter
+    */
+    function revoke(address from) private {
+
+        address representee = from;
+        require(votersData[representee].registered);
+        require(delegations[representee] != representee); // You must have a delegation already
 
-        votersData[nextRepresentative].weight += votersData[nextRepresentee].weight;
-        votersData[nextRepresentee].weight -= votersData[nextRepresentee].weight;
+        address nextRepresentative = delegations[representee];
+
+        // While there is voter that hasn't performed a delegation to another voter
+        // transitively remove weight from their voting power
+        while (delegations[nextRepresentative] != nextRepresentative) {
```

Comment:
I really like the simplicity that comes from thinking of `revoke()` simply as the inverse of `delegate()`. Because of that relationship and therefore the similarity in code, this loop is a piece that we could optimize eventually. That inverse relationship though... I now see what you meant the other day!